src/client/opamCommands.ml

```diff
@@ -2270,8 +2270,9 @@ let repository cli =
       let global = List.mem `Default scope in
       let switches =
         if scope = [] ||
-           List.exists (function `Current_switch | `Switch _ -> true
-                               | _ -> false)
+           List.exists (function
+               | `This_switch | `Current_switch | `Switch _ -> true
+               | _ -> false)
```

Comment:
Side note: we might want to get rid of those dangerous `| _ -> ...` patterns at some point in the future (i.e. enable `-w 4` (fragile pattern matching))